sporkfish/searcher/pvs.py

```diff
@@ -0,0 +1,250 @@
+from typing import Optional, Tuple
+
+import chess
+
+from sporkfish.board.board import Board
+from sporkfish.evaluator.evaluator import Evaluator
+from sporkfish.searcher.minimax import MiniMaxVariants
+from sporkfish.searcher.move_ordering.move_orderer import MoveOrderer
+from sporkfish.searcher.searcher_config import SearcherConfig
+from sporkfish.statistics import NodeTypes
+from sporkfish.zobrist_hasher import ZobristStateInfo
+
+
+class PVSSp(MiniMaxVariants):
+    def __init__(
+        self,
+        evaluator: Evaluator,
+        searcher_config: SearcherConfig = SearcherConfig(),
+    ) -> None:
+        super().__init__(evaluator, searcher_config)
+
+    def _pvs(
+        self,
+        board: Board,
+        depth: int,
+        alpha: float,
+        beta: float,
+        zobrist_state: Optional[ZobristStateInfo],
+    ) -> float:
+        """
+        Principal Variation Search implementation with alpha-beta pruning. For non-root nodes.
```

Comment:
We did implement for root node as well no? just that it falls back to `_quiescence`

Reply:
The root node is the top node, meaning the one we start from in _start_search_from_node.

Reply:
Maybe it is obvious (I don't know much about PVS) - what happens at root node then?

Any minimax like algorithm still need to complete the search to arrive at the best move at root node, right?